webapp/src/components/App/Singlestudy/explore/Modelization/Map/utils.ts

```diff
@@ -1,10 +1,105 @@
-import * as R from "ramda";
+import { AreaNode } from "../../../../../../redux/ducks/studyMaps";
 
-type LinkStyleReturn = [number[], string];
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
 
-export const linkStyle = R.cond<[string], LinkStyleReturn>([
-  [R.equals("dot"), (): LinkStyleReturn => [[1, 5], "round"]],
-  [R.equals("dash"), (): LinkStyleReturn => [[16, 8], "square"]],
-  [R.equals("dotdash"), (): LinkStyleReturn => [[10, 6, 1, 6], "square"]],
-  [R.T, (): LinkStyleReturn => [[0], "butt"]],
-]);
+export type RGB = [number, number, number];
+
+////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////
+
+export const NODE_HEIGHT = 400;
+export const INITIAL_ZOOM = 1;
+export const NODE_COLOR = "rgb(230, 108, 44)";
+
+////////////////////////////////////////////////////////////////
+// Utils
+////////////////////////////////////////////////////////////////
+
+// TODO rework
+export const getNodeWidth = (nodeText: string): number => {
+  const FONT_SIZE = 16;
+  const TEXT_SIZE = nodeText.length;
+
+  if (TEXT_SIZE === 1) {
+    return FONT_SIZE * TEXT_SIZE * 36;
+  }
+  if (TEXT_SIZE <= 2) {
+    return FONT_SIZE * TEXT_SIZE * 20;
+  }
+  if (TEXT_SIZE <= 3) {
+    return FONT_SIZE * TEXT_SIZE * 12;
+  }
+  if (TEXT_SIZE <= 5) {
+    return FONT_SIZE * TEXT_SIZE * 10;
+  }
+  if (TEXT_SIZE <= 6) {
+    return FONT_SIZE * TEXT_SIZE * 8.5;
+  }
+  if (TEXT_SIZE <= 10) {
+    return FONT_SIZE * TEXT_SIZE * 7.5;
+  }
+  return FONT_SIZE * TEXT_SIZE * 6.5;
+};
+
+export function getUpdatedNode(
+  id: string,
+  nodeData: AreaNode[]
+): AreaNode | undefined {
+  return nodeData.find((node) => node.id === id);
+}
+
+const getLuminanace = (values: RGB): number => {
+  const rgb = values.map((v) => {
+    const val = v / 255;
+    return val <= 0.03928 ? val / 12.92 : ((val + 0.055) / 1.055) ** 2.4;
+  });
+  return Number(
+    (0.2126 * rgb[0] + 0.7152 * rgb[1] + 0.0722 * rgb[2]).toFixed(3)
+  );
+};
+
+const getContrastRatio = (colorA: RGB, colorB: RGB): number => {
+  const lumA = getLuminanace(colorA);
+  const lumB = getLuminanace(colorB);
+
+  return (Math.max(lumA, lumB) + 0.05) / (Math.min(lumA, lumB) + 0.05);
+};
+
+export const getTextColor = (bgColor: RGB): string => {
+  const whiteContrast = getContrastRatio(bgColor || [], [255, 255, 255]);
+  const blackContrast = getContrastRatio(bgColor || [], [0, 0, 0]);
+
+  return whiteContrast > blackContrast ? "#ffffff" : "#000000";
+};
+
+////////////////////////////////////////////////////////////////
+// Hooks
+////////////////////////////////////////////////////////////////
+
+/**
+ * Sets the graph nodes from the nodes data
+ */
+export function useRenderNodes(
```

Comment:
The hook has no use here, because you don't memoize the returned value. Change the name of function or use `useMemo` in it.